projects/ssr-tests/src/proxy.utils.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as http from 'http';
+import * as httpProxy from 'http-proxy';
+
+const proxy = (<any>httpProxy).createProxyServer({ secure: false });
+
+/**
+ * Default settings to send http requests.
+ */
+const REQUEST_OPTIONS = {
+  host: 'localhost',
+  port: 4000,
+};
+
+interface ProxyOptions {
+  /**
+   * The url to reroute requests to.
+   */
+  target: string;
+  /**
+   * Number of seconds to delay requests before sending.
+   */
+  delay?: number;
+  /**
+   * Number of status code to set response to.
+   */
+  throwStatus?: number;
+}
+
+/**
+ * Starts an http proxy server on port 9002 with the provided options.
+ */
+export async function startProxyServer(options: ProxyOptions) {
+  return new Promise((resolve) => {
+    const server = http.createServer((req: any, res: any) => {
+      const forwardRequest = () =>
+        proxy.web(req, res, { target: options.target });
+
+      if (options.throwStatus) {
+        proxy.on('proxyRes', (proxyRes: any) => {
+          proxyRes.statusCode = options.throwStatus;
+        });
+      }
+
+      if (options.delay) {
+        setTimeout(forwardRequest, options.delay);
+      } else {
+        forwardRequest();
+      }
+    });
+
+    server.listen(9002, () => {
+      resolve(server);
+    });
+  });
+}
+
+/**
+ * Send an http GET request to a given url.
+ */
+export async function sendRequest(path: string) {
+  return new Promise((resolve, reject) => {
+    const req = http.get({ ...REQUEST_OPTIONS, path }, (res: any) => {
```

Comment:
Let's allow for injecting extra Spartacus config chunk, by passing it via a special cookie. 
We already use this mechanism in our Cypress E2E tests. Let's just use it here as well.
The cookie name is `cxConfigE2E` and the Spartacus `TestConfigModule` reads it and injects
the config passed  via the cookie into Spartacus global config.

Beloww a snippet of code to pass a the cxConfig as an extra cookie:
NOTE: btw. I've changed the signature of the function's argument from `path:string` to object `{path:string, cxConfig?: Config}

```suggestion
import { Config } from '@spartacus/core';
/* ... */

export async function sendRequest({
  path,
  cxConfig
}: {
  path: string,
  cxConfig?: Config
}) {
  return new Promise((resolve, reject) => {
    const req = http.get({ 
      ...REQUEST_OPTIONS, 
      path,
      headers: {
        Cookie: buildCxConfigCookie(cxConfig)
    }, (res: any) => {
```

and let's add a function:

```ts
/**
  * Builds a cookie string with key 'cxConfigE2E' and the given `cxConfig`
  * object as a value that is stringified and URI-encoded.
  *
  * Note: `TestConfigModule` of Spartacus will read this cookie
  * and inject the passed config chunk into Spartacus global config.
  */
function buildCxConfigE2ECookie(cxConfig?: object): string {
  if(!cxConfig) {
    return {};
  }
  const cookieKey = 'cxConfigE2E';
  const cookieValue = encodeURIComponent(JSON.stringify(cxConfig));
  return `${cookieKey}=${cookieValue}`
}
```